lib/class-wp-theme-json-resolver.php

```diff
@@ -186,6 +267,7 @@ private static function get_core_origin() {
 	 */
 	private function get_theme_origin( $theme_support_data = array() ) {
 		$theme_json_data = self::get_from_file( locate_template( 'experimental-theme.json' ) );
+		self::apply_theme_json_translations( $theme_json_data, wp_get_theme()->get_stylesheet() );
```

Comment:
```suggestion
		self::apply_theme_json_translations( $theme_json_data, wp_get_theme()->get( 'TextDomain' ) );
```

That's what `WP_Theme` itself uses for translating theme headers.

Reply:
First need to check if it's not `false` though I think. Without a text domain this function probably should not be called at all.

Reply:
Should we maybe add the textdomain in `theme.json`? Right now we may need `style.css` but in the future it may not necessary

Reply:
Thank you @swissspidy, the suggestion was applied.

@aristath I guess for now we can rely on style.css being preset, but in the future, we probably need to add a text-domain and many other things, that are currently part of style.css.